### PR TITLE
[ONNX] Add ArgMin/Max operators

### PR DESCRIPTION
**Pull request by postrational**

### CODE REVIEW

File: src/ngraph/frontend/onnx_import/op/argmin.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/argmin.hpp"
+#include "ngraph/node.hpp"
+#include "ngraph/node_vector.hpp"
+#include "ngraph/op/reshape.hpp"
+
```
Comment (rkimballn1) [inline src/ngraph/frontend/onnx_import/op/argmin.cpp]: remove line

File: src/ngraph/frontend/onnx_import/op/argmin.hpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include "ngraph/node_vector.hpp"
+
```
Comment (rkimballn1) [inline src/ngraph/frontend/onnx_import/op/argmin.hpp]: remove line

**Comment by rkimballn1** [comment]
I approved by I did have a couple of very minor changes

File: src/ngraph/frontend/onnx_import/op/argmin.cpp (hunk 1 above)
Comment (postrational) [reply src/ngraph/frontend/onnx_import/op/argmin.cpp]: I wanted to keep core nGraph includes separate from ONNX importer includes. Do you think that's a bad idea?

File: src/ngraph/frontend/onnx_import/op/argmin.cpp [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+#include "core/attribute.hpp"
+#include "core/node.hpp"
+#include "utils/reduction.hpp"
+#include "utils/reshape.hpp"
```
Comment (arogowie-intel) [inline src/ngraph/frontend/onnx_import/op/argmin.cpp]: Please remove unused headers.

File: src/ngraph/frontend/onnx_import/op/argmin.cpp [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+{
+    namespace onnx_import
+    {
+        namespace op
+        {
+            namespace set_1
+            {
+                NodeVector argmin(const Node& node)
+                {
+                    return {reduction::make_ng_arg_reduction_op<ngraph::op::ArgMin>(node)};
```
Comment (arogowie-intel) [inline src/ngraph/frontend/onnx_import/op/argmin.cpp]: Consider moving this into `*.hpp` file.

File: src/ngraph/frontend/onnx_import/utils/reduction.hpp [hunk 5]
```diff
@@ -101,6 +102,38 @@ namespace ngraph
                     reshape::get_default_axis_vector(op_node->get_shape().size()),
                     Shape{output_shape});
             }
+
+            template <class T>
+            std::shared_ptr<ngraph::Node> make_ng_arg_reduction_op(const Node& node)
```
Comment (arogowie-intel) [inline src/ngraph/frontend/onnx_import/utils/reduction.hpp]: Ngraph is naming this class of operations as `IndexReduction` - this operators inherits from this class. Consider renaming it to sth  _index_-like.

File: test/onnx_import.cpp [hunk 6]
```diff
@@ -74,6 +75,24 @@ TEST(onnx, model_addmul_abc)
...
+    auto function = onnx_import::import_onnx_function(
+        file_util::path_join(SERIALIZED_ZOO, "onnx/argmin_no_keepdims.onnx"));
+
+    auto backend = runtime::Backend::create("INTERPRETER");
+
+    auto input_tensor = backend->create_tensor(element::f32, input_shape);
+    copy_data(input_tensor, std::vector<float>{2, 1, 3, 10});
+    auto result_tensor = backend->create_tensor(element::i64, result_shape);
+
+    backend->call_with_validate(function, {result_tensor}, {input_tensor});
```
Comment (arogowie-intel) [inline test/onnx_import.cpp]: Why you're not using `execute` function? It's capable to handle different types for input and output tensors. Look here https://github.com/NervanaSystems/ngraph/blob/01205771e8b12024a013dc1f87fe4305327dab46/test/onnx_import.cpp#L972.

**Pull request merged**